MNT: update repository

kmuehlbauer opened the pull request:
- remove Python 2 compatibility
- switch to `h5netcdf` org
- switch from travis to github actions
- use black style, isort and flake8 testing
- test both h5py 2 and 3

kmuehlbauer commented:
@shoyer This is the first shot which works quite nice in [my fork](https://github.com/kmuehlbauer/h5netcdf/actions). Please have a look over there before merging. I've taken the chance to also add checks for lint and style (needed to reformat files). Hope this fits your requirements.

After this is merged I'll update my other PR to get h5py>=3 resolved.

shoyer commented on h5netcdf/legacyapi.py:
```diff
@@ -1,7 +1,8 @@
 from . import core
-from .compat import unicode
+
+# from .compat import unicode
```
Delete?

kmuehlbauer replied:
Sure!

shoyer commented on h5netcdf/legacyapi.py:
```diff
@@ -31,32 +33,34 @@ def __setattr__(self, name, value):
-        if h5py.check_dtype(vlen=dt) is unicode:
-            return str
+        # if h5py.check_dtype(vlen=dt) is unicode:
+        #    return str
```
I think we still want this for compatibility with netCDF4-python?

kmuehlbauer replied:
Yes, forgot to fix this one...

shoyer commented on h5netcdf/tests/test_h5netcdf.py:
```diff
@@ -101,223 +111,234 @@ def is_h5py_char_working(tmp_netcdf, name):
-    assert v[0] == u'foo'
+    v = ds.variables["var_len_str"]
+    print("dtype:", v.dtype)
+    # assert v.dtype == 'O'
```
This is the test for the commented out inline setting "str" as a dtype

kmuehlbauer replied:
And forgot this one too...

shoyer reviewed (commented):
Thanks for cleanup PR!

kmuehlbauer commented:
@shoyer I've corrected the things per your suggestion. Removed the h5py3 fix. Added h5py3 tests to the matrix (to see things breaking).

I think this now ready for review/merge and we can move on to #82.

shoyer commented:
thanks!
